src/jit/copyprop.cpp

```diff
@@ -259,14 +266,15 @@ void Compiler::optCopyProp(BasicBlock* block, GenTreePtr stmt, GenTreePtr tree,
         lvaTable[newLclNum].incRefCnts(block->getBBWeight(this), this);
         tree->gtLclVarCommon.SetLclNum(newLclNum);
         tree->AsLclVarCommon()->SetSsaNum(newSsaNum);
+        gtUpdateSideEffects(stmt, tree);
 #ifdef DEBUG
         if (verbose)
         {
             printf("copy propagated to:\n");
             gtDispTree(tree, nullptr, nullptr, true);
         }
 #endif
-        break;
+        return;
```

Comment:
This change looks odd, considering that the function ends right after the loop and there's even a useless `return` after the loop.